subsys/net/lib/sockets/sockets.c

```diff
@@ -459,6 +459,107 @@ Z_SYSCALL_HANDLER(zsock_sendto, sock, buf, len, flags, dest_addr, addrlen)
 }
 #endif /* CONFIG_USERSPACE */
 
+static int sock_get_pkt_src_addr(struct net_pkt *pkt,
+				 enum net_ip_protocol proto,
+				 struct sockaddr *addr,
+				 socklen_t addrlen)
+{
+	int ret = 0;
+	struct net_pkt_cursor backup;
+	u16_t *port;
+
+	if (!addr || !pkt) {
+		return -EINVAL;
+	}
+
+	net_pkt_cursor_backup(pkt, &backup);
+	net_pkt_cursor_init(pkt);
```

Comment:
So, this is first time I see this "packet cursor" thing (I like the name, hope the implementation is unbloated). So, I know nothing about it. But I find strange that we first backup packet cursor, then initialize it. Either they're swapped, or there's naming problem, e.g. "net_pkt_cursor_init" is actually "net_pkt_cursor_reset" (i.e. resets cursor to the beginning of a packet).

Reply:
I am not going to do any change name right now. semantically it's not that much different init vs reset.

both mean the cursor is set at the beginning of net_pkt buffer (if there is one).